### PR TITLE
Use registry templates to create new app

### PR DESCRIPTION
**Pull request by xdoardo**
As per title, this PR removes the old templating mechanism and institutes a new one using the registry templates, solving #4578. 

Mandatory screenshot: 
image

Side quests: 
- Passes the owner information from `wasmer deploy` to `wasmer app create` (solves #4701)
- Correctly serializes the package information when creating from an existing manifest (solves #4674)
- Implicitly solves #4605, but introduces a potentially breaking change: `new-package-name` is not used anymore and, as of now, the `static-website` template is not available from the registry (with that name, at least).

### CODE REVIEW

File: lib/cli/src/utils/prompts.rs [hunk 1]
```diff
@@ -23,38 +22,38 @@ pub fn prompt_for_ident(message: &str, default: Option<&str>) -> Result<String,
     }
 }
 
-/// Ask a user for a package name.
-///
-/// Will continue looping until the user provides a valid name.
-pub fn prompt_for_package_ident(
-    message: &str,
-    default: Option<&str>,
-) -> Result<NamedPackageIdent, anyhow::Error> {
```
Comment (syrusakbary) [inline lib/cli/src/utils/prompts.rs:32]: Why we are not prompting anymore for a package?

File: lib/cli/src/utils/mod.rs [hunk 2]
```diff
@@ -111,79 +108,79 @@ pub fn load_package_manifest(
...
-                        break Ok((name, None));
-                    } else {
-                        eprintln!("Package '{name}' already exists");
-                    }
-                }
-            }
-        }
-    }
-}
+// /// Defines how to check for a package.
```
Comment (syrusakbary) [inline lib/cli/src/utils/mod.rs:135]: Remove me if you are not going to use it

File: lib/cli/src/utils/mod.rs [hunk 3]
```diff
@@ -111,79 +108,79 @@ pub fn load_package_manifest(
...
-
-/// Defines how to check for a package.
-pub enum PackageCheckMode {
-    /// The package must exist in the registry.
-    MustExist,
-    /// The package must NOT exist in the registry.
-    #[allow(dead_code)]
-    MustNotExist,
-}
+///// Ask a user for a package name.
```
Comment (syrusakbary) [inline lib/cli/src/utils/mod.rs:111]: Remove me if not longer needed

**Comment by theduke** [comment]
Do we have a static site template?
Should have that before releasing the the version.

**Review by syrusakbary — APPROVED** [review]

**Pull request merged**